ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.component.js

```diff
@@ -175,20 +185,18 @@ export default class ConfirmAddSuggestedToken extends Component {
   }
 
   /**
-   * Returns true if any pendingTokens both:
+   * Returns true if any suggestedAssets both:
    * - Share a symbol with an existing `tokens` member.
    * - Does not share an address with that same `tokens` member.
    * This should be flagged as possibly deceptive or confusing.
    */
-  checkNameReuse(pendingTokens, tokens) {
-    const duplicates = Object.keys(pendingTokens)
-      .map((addr) => pendingTokens[addr])
-      .filter((token) => {
-        const dupes = tokens
-          .filter((old) => old.symbol === token.symbol)
-          .filter((old) => old.address !== token.address);
-        return dupes.length > 0;
-      });
+  checkNameReuse(suggestedAssets, tokens) {
+    const duplicates = suggestedAssets.filter(({ asset }) => {
+      const dupes = tokens
+        .filter((old) => old.symbol === asset.symbol)
```

Comment:
Doing two consecutive filters can be expensive -- best to combine them into one.

Reply:
✅